addons/semicolon/semicolon.js

```diff
@@ -1,14 +1,14 @@
 export default async function ({ addon, global, console }) {
-  if (document.body.contains(document.querySelector("#signature"))) {
-    console.log("Signature already existed");
-  } else {
-    if (document.location.href.split("/")[5] == "editor") {
-      console.log("In editor");
+    if (document.body.contains(document.querySelector("#signature"))) {
+      console.log("Signature already exists!");
     } else {
-      var semicolon = document.createElement("p");
-      semicolon.textContent = ";";
-      semicolon.classList.add("semicolon");
-      document.body.appendChild(semicolon);
+      if (addon.tab.editorMode === "editor") {
```

Comment:
You might want to use the same pattern some addons use to re-inject the semicolon if the user goes outside the editor.